src/Datadog.Trace/Propagation/ContextPropagatorBuilder.cs

```diff
@@ -15,14 +16,19 @@ internal static class ContextPropagatorBuilder
                 { PropagatorType.Default, (convention) => new DDSpanContextPropagator(convention) },
             };
 
-        public static IPropagator BuildPropagator(PropagatorType propagator, ITraceIdConvention traceIdConvention)
+        public static ICollection<IPropagator> BuildPropagators(IEnumerable<PropagatorType> propagatorTypes, ITraceIdConvention traceIdConvention)
         {
-            if (_propagatorSelector.TryGetValue(propagator, out Func<ITraceIdConvention, IPropagator> getter))
+            return propagatorTypes.Select(type => BuildPropagator(type, traceIdConvention)).ToList();
```

Comment:
Any reason why `ToList()` and not `ToArray()` ?

Reply:
Not sure there is much difference. In nano world over the time one used to be "better" than another and vice versa.
https://stackoverflow.com/questions/1105990/is-it-better-to-call-tolist-or-toarray-in-linq-queries

Reply:
resolve